pkg/client/api/auth/auth.go

```diff
@@ -129,9 +130,9 @@ func (c *auth) ClientCredentialsToken(authKind, slugID, clientID, clientSecret s
 }
 
 func (c *auth) deviceAuthURL(authKind, authProvider string) string {
-	return fmt.Sprintf("%s%s/%s/%s/%s/%s", c.client.BaseURL.String(), utils.APIPrefix, utils.PathAuthProviders, authKind, authProvider, utils.PathDeviceCode)
+	return urlutils.CloneURLAddingPath(c.client.BaseURL, utils.APIPrefix, utils.PathAuthProviders, authKind, authProvider, utils.PathDeviceCode).String()
```

Comment:
I don't get why it is fixing the issue ? For me it doesn't change anything excepting adding more complexity on how we are managing URL currently.

Reply:
.String() was giving a trailing slash, and APIPrefix was containing a starting slash so it was doubling the slash. In the clone function I'm using a mechanism used heavily already in the rest of the code so I mutualized